File: src/ru/ifmo/ctd/evdokimov/simple/ByteCodeLoader.java
package ru.ifmo.ctd.evdokimov.simple;

public class ByteCodeLoader extends ClassLoader {
    public static final ByteCodeLoader clazz = new ByteCodeLoader();

    public Class<?> loadClass(byte[] bytecode) {
        return defineClass(null, bytecode, 0, bytecode.length);
    }
}

File: src/ru/ifmo/ctd/evdokimov/alang/ByteCodeLoader.kt
package ru.ifmo.ctd.evdokimov.alang

class ByteCodeLoader : ClassLoader() {

    fun loadClass(bytecode: ByteArray): Class<*> {
        return defineClass(null, bytecode, 0, bytecode.size)
    }

    companion object {
        val clazz = ByteCodeLoader()
    }
}